apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -103,6 +105,35 @@ export class AppRoomBridge extends RoomBridge {
 		return this.orch.getConverters()?.get('users').convertById(room.u._id);
 	}
 
+	protected async getMessages(roomId: string, options: GetMessagesOptions, appId: string): Promise<IMessageRaw[]> {
+		this.orch.debugLog(`The App ${appId} is getting the messages of the room: "${roomId}" with options:`, options);
+
+		const { limit, skip = 0, sort } = options;
+
+		const messageConverter = this.orch.getConverters()?.get('messages');
+		if (!messageConverter) {
+			throw new Error('Message converter not found');
+		}
+
+		const messageQueryOptions: FindOptions<ICoreMessage> = {
+			limit,
+			skip,
+			sort,
+		};
+
+		const query = {
```

Comment:
Just being sure we have an index covering this and it won't get to the `rid` index

Reply:
Yes, we have an index that covers this query. The existing compound index `{ rid: 1, t: 1, 'u._id': 1 }` is being used effectively for this query.

The `executionStatus` for a sample Query.
```
    winningPlan: {
      stage: 'FETCH',
      filter: {
        '$and': [
          {
            t: {
              '$not': {
                '$exists': true
              }
            }
          },
          {
            _hidden: {
              '$not': {
                '$eq': true
              }
            }
          }
        ]
      },
      inputStage: {
        stage: 'IXSCAN',
        keyPattern: {
          rid: 1,
          t: 1,
          'u._id': 1
        },
        indexName: 'rid_1_t_1_u._id_1',
        isMultiKey: false,
        multiKeyPaths: {
          rid: [],
          t: [],
          'u._id': []
        },
        isUnique: false,
        isSparse: false,
        isPartial: false,
        indexVersion: 2,
        direction: 'forward',
        indexBounds: {
          rid: [
            '["GENERAL", "GENERAL"]'
          ],
          t: [
            '[null, null]'
          ],
          'u._id': [
            '[MinKey, MaxKey]'
          ]
        }
      }
    },
```

Reply:
Execution time on >3M messages room.

```
executionStats: {
    executionSuccess: true,
    nReturned: 100,
    executionTimeMillis: 4,
    totalKeysExamined: 100,
    totalDocsExamined: 100,
```

That was fast. Thanks to Kev's nice script, it was easy to fill a room with messages.